src/client.js

```diff
@@ -30,12 +33,28 @@ class StitchClient {
 
     this.clientAppID = clientAppID;
 
-    this.appUrl = `${baseUrl}/api/public/v1.0`;
-    this.authUrl = `${baseUrl}/api/public/v1.0/auth`;
-    if (clientAppID) {
-      this.appUrl = `${baseUrl}/api/client/v1.0/app/${clientAppID}`;
-      this.authUrl = `${this.appUrl}/auth`;
-    }
+    this.authUrl = (
+      clientAppID ?
+        `${baseUrl}/api/client/v1.0/app/${clientAppID}/auth` :
+        `${baseUrl}/api/public/v1.0/auth`
+    );
+
+    this.rootURLsByAPIVersion = {
+      [v1]: {
+        public: `${baseUrl}/api/public/v1.0`,
+        client: `${baseUrl}/api/client/v1.0`,
+        app: (clientAppID ?
+              `${baseUrl}/api/client/v1.0/app/${clientAppID}` :
+              `${baseUrl}/api/public/v1.0`)
+      },
+      [v2]: {
+        public: `${baseUrl}/api/public/v2.0`,
+        client: `${baseUrl}/api/client/v2.0`,
```

Comment:
[q] Should we bother including the two client references here when the API doesn't actually exist?

Reply:
Well, there's no way to actually use the client v2 API from the SDK right now anyway, since its methods (executePipeline, for instance) still fall using v1, so I don't think it matters; once the API does exist though, we're going to need to add this anyway, so might as well just include it now.

Reply:
Sounds good